src/Backend.cpp

```diff
@@ -135,7 +135,8 @@ void HABackend::threadrunner()
         // cerr<<j.dump()<<endl;
         for (auto& [domain, _services] : j["result"].items()) {
           // cerr<<service.dump()<<endl;
-          domains[domain] = std::make_shared<HADomain>(_services);
+          auto entityType = magic_enum::enum_cast<EntityType>(domain, magic_enum::case_insensitive).value_or(EntityType::OTHER);
```

Comment:
this assumes we have a maximum of one OTHER

Reply:
Not sure what you mean, but to me 'OTHER' means we don't support it.

The enum at the moment has a few other values, but i think in the end that enum should only have things we support, because somewhere in our code we need to do 'if domain == enum::bla' and then create the right HAEntity and/or UI..

Apart from getting away with string comparisons, i think the enum does this as well for us, be clear what we support.

Reply:
ah, I actually meant the line just below (line 139). If HA sends us Light, we store it in `domains[Light]`. If HA sends us Climate, we store it in `domains[Climate]`. If HA sends us Button, we store it in `domains[OTHER]`. If HA then sends us Fan, we store it in `domains[OTHER]`, losing Button. There's likely a better way..

Or it might just be simpler to use a string as the key for `domains[]` even though we have the enum?

Reply:
Note that I'm fine with writing this in a FIXME comment and leaving it for now